pkg/services/sqlstore/datasource.go

```diff
@@ -66,11 +66,20 @@ func GetDataSource(query *models.GetDataSourceQuery) error {
 
 func GetDataSources(query *models.GetDataSourcesQuery) error {
 	var sess *xorm.Session
-	if query.DataSourceLimit <= 0 {
-		sess = x.Where("org_id=?", query.OrgId).Asc("name")
-	} else {
-		sess = x.Limit(query.DataSourceLimit, 0).Where("org_id=?", query.OrgId).Asc("name")
+	sess = x.Asc("name")
+
+	if query.OrgId != 0 {
```

Comment:
There's a potential problem with this one if orgId would be zero and used from an inappropriate place like listing data sources in the API, perhaps by some bug. That's why I would recommend having a separate GetDataSourcesByTypeQuery which makes things more explicit. But I'm open to discuss it.

Reply:
I see your point. changed it :)